python/cudf/dataframe/index.py

```diff
@@ -71,8 +73,8 @@ def take(self, indices):
             return RangeIndex(indices.size)
         else:
             # Gather
-            index = cudautils.gather(data=self.gpu_values, index=indices)
-            col = self.as_column().replace(data=Buffer(index))
+            index = cpp_copying.apply_gather_array(self.gpu_values, indices)
+            col = self.as_column().replace(data=index.data)
             return as_index(col)
```

Comment:
Can just use a gather column instead of gathering arrays and recreating columns.